k2/csrc/fsa_equivalent.h

```diff
@@ -22,13 +23,58 @@ namespace k2 {
  */
 bool IsRandEquivalent(const Fsa &a, const Fsa &b, std::size_t npath = 100);
 
+/*
+  Returns true if the Fsa `a` is stochastically equivalent to `b` by randomly
+  generating `npath` paths from one of them and then checking if each path exist
+  in the other one and the sum of weights along that path are the same.
+ */
+template <class FbWeightType>
```

Comment:
It should be

```cpp
template<FbWeightType Type>
```

And you can invoke it with `IsRandEquivalent<kMaxWeight>(....)`
or `IsRandEquivalent<kLogSumWeight>(....)`


The same applies to the subsequent `ShortestDistance `.

Reply:
Ah, thanks, that's what I wanted to do. Will change